openwisp_controller/geo/tests/test_api.py

```diff
@@ -655,6 +655,33 @@ def test_create_devicelocation_outdoor_location_with_floorplan(self):
         )
         self.assertEqual(self.object_location_model.objects.count(), 0)
 
+    def test_endpoints_for_non_existent_device(self):
+        device_id = uuid.uuid4()
+        floorplan = self._create_floorplan()
+        location = floorplan.location
+        url = reverse('geo_api:device_location', args=[device_id])
+
+        with self.subTest('Retrieve operation'):
+            response = self.client.get(
+                url,
+            )
+            self.assertEqual(response.status_code, 404)
+
+        with self.subTest('Create operation'):
```

Comment:
Shouldn't it be an update?
```suggestion
        with self.subTest('Udpate operation'):
```